src/data/list/basic.lean

```diff
@@ -3009,8 +3009,63 @@ by convert filter_eq_nil.2 (λ _ _, id)
       take_while_append_drop l]
     else by rw [take_while, drop_while, if_neg pa, if_neg pa, nil_append]
 
+@[simp] theorem length_filter_le (l : list α) : (filter p l).length ≤ l.length :=
+begin
+  induction l;
+  rw list.filter;
+  simp,
+  cases classical.em (p l_hd);
+  simp [h],
+  {
+    exact l_ih,
+  },
+  {
+    exact le_add_right l_ih
+  }
+end
+
+@[simp] theorem length_filter_lt_of_mem {a : α} (l : list α) : 
+  a ∈ l → ¬p a → (filter p l).length < l.length :=
+begin
+  induction l;
+  simp,
+  intros h_in hpa,
```

Comment:
```suggestion
@[simp] theorem length_filter_lt_of_mem {a : α} (l : list α) (ha : a ∈ l) (hpa : ¬p a) : (filter p l).length < l.length :=
begin
  induction l;
  simp,
```
It's usually better to put hypotheses before the colon when possible.

Reply:
Sure, done.
 I assume that `mem_filter_of_mem` (and a few other theorems) use implications like this for syntax convenience from pattern matching? Or is there some other reason?